### PR TITLE
Add Nuget.Packaging 6.11.0 and dependencies

### PR DESCRIPTION
**Pull request by MilenaHristova**
Issue: https://github.com/dotnet/source-build/issues/4571

I had to do a few edits to make the projects build, I found https://github.com/dotnet/source-build/issues/4328  and https://github.com/dotnet/source-build-reference-packages/pull/861#issuecomment-1882000238 and did the same modifications

### CODE REVIEW

File: src/referencePackages/src/system.formats.asn1/6.0.0/System.Formats.Asn1.6.0.0.csproj [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (ellahathaway) [inline src/referencePackages/src/system.formats.asn1/6.0.0/System.Formats.Asn1.6.0.0.csproj:1]: System.Formats.Asn1.6.0.0 contains a vulnerability, so we can't add it to SBRP (see https://github.com/dotnet/source-build/issues/4518). If there's a reference to this package, you can manually change the reference to 6.0.1 and add a comment above it saying that it was changed manually (see https://github.com/dotnet/source-build-reference-packages/pull/992/files)

**Review by ellahathaway — CHANGES_REQUESTED** [review]

File: src/referencePackages/src/system.formats.asn1/6.0.0/System.Formats.Asn1.6.0.0.csproj (hunk 1 above)
Comment (MilenaHristova) [reply src/referencePackages/src/system.formats.asn1/6.0.0/System.Formats.Asn1.6.0.0.csproj]: thanks, I removed it

**Comment by ellahathaway** [comment]
You'll have to add these packages as well:

NuGet.Commands.6.11.0
NuGet.Credentials.6.11.0
NuGet.DependencyResolver.Core.6.11.0
NuGet.LibraryModel.6.11.0
NuGet.ProjectModel.6.11.0
NuGet.Protocol.6.11.0

There should be 11 total packages being upgraded/added.

**Review by mthalman — APPROVED** [review]

**Review by ellahathaway — APPROVED** [review]

**Comment by ellahathaway** [comment]
Merging to unblock builds and other PRs

**Pull request merged**